recipes/boost/all/rebuild-dependencies.py

```diff
@@ -0,0 +1,268 @@
+import argparse
+import dataclasses
+from pathlib import Path
+import re
+import subprocess
+import sys
+import tempfile
+from typing import Dict, List
+
+from conans import tools
+import json
+import yaml
+
+
+BOOST_GIT_URL = "https://github.com/boostorg/boost.git"
+
+
+@dataclasses.dataclass
+class BoostDependenciesExport(object):
+    version: str
+    dependencies: Dict[str, List[str]] = dataclasses.field(default_factory=dict)
+    libs: Dict[str, List[str]] = dataclasses.field(default_factory=dict)
+
+
+@dataclasses.dataclass
+class BoostDependencies(object):
+    buildables: List[str]
+    export: BoostDependenciesExport
+
+
+class BoostDependencyBuilder(object):
+    def __init__(self, boost_version: str, boostdep_version: str, tmppath: Path, git_url: str, outputdir: Path, unsafe: bool):
+        self.boost_version = boost_version
+        self.boostdep_version = boostdep_version
+        self.git_url = git_url
+        self.tmppath = tmppath
+        self.outputdir = outputdir
+        self.unsafe = unsafe
+
+    @property
+    def boost_path(self) -> Path:
```

Comment:
```suggestion
    def boost_path(self):
```

type hints are not supported in 2.7

Reply:
This script is meant to be run occasionally on a person modifying the boost recipe.
Right now, it's even not in `exports` or `export_sources`.

I think it's safe to assume now that every conan recipe developer has a python3 release with type hints.

Reply:
Agreed!